Implemented revoke token in RedditOAuthAgent

HeroesHannibal opened the pull request:
We implemented the revoke() function within the RedditOAuthAgent class. We've provided examples of how to revoke a token in the example files and we've written JUnit tests for the implemented functionality.

Commit:
Implemented revoke_token method.

Added code to post to https://www.reddit.com/api/v1/revoke_token in
order to revoke the refresh_token or access_token depending on which is
required.

Commit:
Refactored code to fix long method bad smell.

The creation of the builder in the revoke method was found to be
excessively long. It was extracted into a new method called
createBuilder to carry out the same task. The ExampleVotingRequest has
been run again to ensure that the functionality was unaffected.

HeroesHannibal committed:
Cleaned up code and updated examples to revoke token after use

HeroesHannibal committed:
Added tests

HeroesHannibal committed:
Corrected Indentation

snkas commented:
Looks good in general, I don't have the time to pull and validate the method. Added small suggestion. Also add a test when the method fails and for the execution branches created by various input values?

HeroesHannibal committed:
Added additional test case for failure scneario

HeroesHannibal commented:
Hey! Thanks for reviewing this. I've made the changes you requested. For the tests, there are only really two test cases here since the api itself only returns a 401 if client authentication was incorrect or 202 in every other case, even if the token is not valid, or if it was revoked multiple times.

Commit:
Read Me for TAs

In compliance with the instructor’s note to include a read me for the
TAs, pushing a txt document that outlines the enhancements we made for
this project.
